tests/test_binning.py

```diff
@@ -199,44 +201,105 @@ def test_bin_edges_to_bin_centers():
+        bin_centers.append(
+            np.linspace(args[2][i][0], args[2][i][1], args[0][i] + 1),
+        )
+
+    if arg_type == "int":
+        bins_ = args[0][0]
+        axes_ = args[1]
+        ranges_ = args[2]
+        bins_expected = [bins_] * len(args[0])
+        axes_expected = axes_
+        ranges_expected = ranges_
+    elif arg_type == "list_int":
+        bins_ = args[0]
+        axes_ = args[1]
+        ranges_ = args[2]
+        bins_expected = bins_
+        axes_expected = axes_
+        ranges_expected = ranges_
+    elif arg_type == "array":
+        bins_ = []
+        for i in range(len(args[0])):
+            bins_.append(bin_centers[i])
+        axes_ = args[1]
+        bins_expected = bins_
+        axes_expected = axes_
+    elif arg_type == "tuple":
+        bins_ = []
+        for i in range(len(args[0])):
+            bins_.append((args[2][i][0], args[2][i][1], args[0][i]))
+        axes_ = args[1]
+        bins_expected = args[0]
+        axes_expected = axes_
+        ranges_expected = args[2]
+    elif arg_type == "dict_int":
+        bins_ = {}
+        for i, axis in enumerate(args[1]):
+            bins_[axis] = args[0][i]
+        ranges_ = args[2]
+        bins_expected = args[0]
+        axes_expected = args[1]
+        ranges_expected = args[2]
+    elif arg_type == "dict_array":
+        bins_ = {}
+        for i, axis in enumerate(args[1]):
+            bins_[axis] = bin_centers[i]
+        bins_expected = bin_centers
+        axes_expected = args[1]
+    elif arg_type == "dict_tuple":
+        bins_ = {}
+        for i, axis in enumerate(args[1]):
+            bins_[axis] = (args[2][i][0], args[2][i][1], args[0][i])
+        bins_expected = args[0]
+        axes_expected = args[1]
+        ranges_expected = args[2]
+
+    bins__, axes__, ranges__ = simplify_binning_arguments(
+        bins_,
+        axes_,
+        ranges_,
     )
-    assert bins_ == [bins[0]]
-    assert axes_ == [axes[0]]
-    assert ranges_ == (ranges[0],)
-
-
-def test_simplify_binning_arguments_edges():
-    """Test the result of the _simplify_binning_arguments functions for bin edges"""
-    bin_edges = [np.linspace(r[0], r[1], b) for r, b in zip(ranges, bins)]
-    bin_edges_, axes_, ranges_ = _simplify_binning_arguments(bin_edges, axes)
-    for bin_, bin_edges_ in zip(bin_edges_, bin_edges):
-        np.testing.assert_allclose(bin_, bin_edges_)
-    assert axes_ == axes
-    assert ranges_ is None
-
-
-def test_simplify_binning_arguments_tuple():
-    """Test the result of the _simplify_binning_arguments functions for bin tuples"""
-    bin_tuple = [tuple((r[0], r[1], b)) for r, b in zip(ranges, bins)]
-    bins_, axes_, ranges_ = _simplify_binning_arguments(bin_tuple, axes)
-    assert bins_ == bins
-    assert axes_ == axes
-    assert ranges_ == ranges
+    #    for (bin_, bin__) in zip(bins__, bin_centers):
+    #        np.testing.assert_array_equal(bin_, bin__)
+    assert bins__ == bins_expected
```

Comment:
why not use `np.testing.assert_allclose` here? or if they need to be identical: `np.testing.assert_array_equal`

Reply:
They are not arrays, but lists of int or np.array at that point. Not sure if that will work, but I can try. Also not sure what the == actually tests, though.

Reply:
Updated to assert_array_equals now